x-pack/metricbeat/module/aws/cloudwatch/cloudwatch.go

```diff
@@ -189,14 +202,51 @@ func (m *MetricSet) Fetch(report mb.ReporterV2) error {
 
 		for namespace, namespaceDetails := range namespaceDetailTotal {
 			m.logger.Debugf("Collected metrics from namespace %s", namespace)
-
 			// filter listMetricsOutput by detailed configuration per each namespace
 			filteredMetricWithStatsTotal := filterListMetricsOutput(listMetricsOutput, namespace, namespaceDetails)
 
 			// get resource type filters and tags filters for each namespace
 			resourceTypeTagFilters := constructTagsFilters(namespaceDetails)
 
-			eventsWithIdentifier, err := m.createEvents(svcCloudwatch, svcResourceAPI, filteredMetricWithStatsTotal, resourceTypeTagFilters, regionName, startTime, endTime)
+			//Check whether namespace is APIGW
+			var (
+				checkns_lower            = strings.ToLower(checkns)
+				checkresource_type_lower = strings.ToLower(checkresource_type)
```

Comment:
These can be moved to var definitions at line 29 of this file :) This avoids unnecessary var creation inside the loop that starts at line 203

Reply:
yes done in [535b905](https://github.com/elastic/beats/pull/40755/commits/535b9059fa11776e454bfd89742c7e9cdd78b555)